### PR TITLE
Cleanup and refactor

### PR DESCRIPTION
**Pull request by fabriziosestito**

### CODE REVIEW

File: elixir/lib/contracts.ex [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  @moduledoc """
+  This facade contains functions to eoncode/decode events to/from protobuf
+  """
+
+  require Logger
+
+  @doc """
+  Encode and wrap an event struct to a protobuf CloudEvent.
+  """
+  @spec to_event(struct(), Keyword.t()) :: binary()
```
Comment (dottorblaster) [inline elixir/lib/contracts.ex:11]: So the API would be

```elixir
Trento.Contracts.to_event(%Trento.Contracts.GatheredFacts{...})
```

right?

**Review by dottorblaster — APPROVED** [review]
Mega-LGTM

File: elixir/lib/contracts.ex (hunk 1 above)
Comment (fabriziosestito) [reply elixir/lib/contracts.ex:11]: yes

```elixir
%{params}
|> GatheredFacts.new()
|> Trento.Contracts.to_event()
```

Comment (dottorblaster) [reply elixir/lib/contracts.ex:11]: image

**Review by CDimonaco — APPROVED** [review]
image
LGTM

**Pull request merged**